webpack/assets/javascripts/react_app/redux/reducers/notifications/index.js

```diff
@@ -32,14 +32,21 @@ const hasUnreadMessages = notifications => {
 
 export default (state = initialState, action) => {
   const { payload } = action;
+  const { SUCCESS, FAILURE } = actionTypeGenerator(NOTIFICATIONS);
 
   switch (action.type) {
-    case NOTIFICATIONS_POLLING_STARTED:
-      return state.set('isPolling', true);
-    case NOTIFICATIONS_GET_NOTIFICATIONS:
-      return state
-        .set('notifications', payload.notifications)
-        .set('hasUnreadMessages', hasUnreadMessages(payload.notifications));
+    case SUCCESS:
+      return state.merge({
+        notifications: payload.notifications,
+        hasUnreadMessages: hasUnreadMessages(payload.notifications),
+      });
+    case FAILURE: {
+      const { error } = payload;
+      if (error.response && error.response.status === 401) {
+        redirectToLogin();
+      }
```

Comment:
The reducer shouldn't handle that, is there an option to move it? middleware/action would be better place

Reply:
moved it into to the component file, thanks